src/inet/UDPEndPointImpl_OpenThread.cpp

```
@@ -0,0 +1,268 @@
+
+    otUdpOpen(mOTInstance, &socket, handleUdpReceive, this);
+    otUdpBind(mOTInstance, &socket, &listenSockAddr, OT_NETIF_THREAD);
+
+    return chip::DeviceLayer::Internal::MapOpenThreadError(err);
+}
+
+CHIP_ERROR UDPEndPointImplOT::BindImpl(IPAddressType addressType, const IPAddress & addr, uint16_t port, InterfaceId interface)
+{
+
+    if (addressType == IPAddressType::kIPv6)
+    {
+        ReturnErrorOnFailure(IPv6Bind(mSocket, addr, port, interface));
+    }
+    else
+    {
+        return INET_ERROR_WRONG_ADDRESS_TYPE;
+    }
+
+    mBoundPort   = port;
+    mBoundIntfId = interface;
+
+    return CHIP_NO_ERROR;
+}
+
+InterfaceId UDPEndPointImplOT::GetBoundInterface() const
+{
+    return mBoundIntfId;
+}
+
+uint16_t UDPEndPointImplOT::GetBoundPort() const
+{
+    return mBoundPort;
+}
+
+CHIP_ERROR UDPEndPointImplOT::ListenImpl()
+{
+    // Nothing to do. Callback was set upon Binding call.
+    return CHIP_NO_ERROR;
+}
+
+void UDPEndPointImplOT::HandleDataReceived(System::PacketBufferHandle && msg)
+{
+    if ((mState == State::kListening) && (OnMessageReceived != nullptr))
+    {
+        const IPPacketInfo * pktInfo = GetPacketInfo(msg);
+
+        if (pktInfo != nullptr)
+        {
+            const IPPacketInfo pktInfoCopy = *pktInfo; // copy the address info so that the app can free the
+                                                       // PacketBuffer without affecting access to address info.
+
+            msg->ConsumeHead(sizeof(IPPacketInfo));
+            OnMessageReceived(this, std::move(msg), &pktInfoCopy);
+        }
+        else
+        {
+            if (OnReceiveError != nullptr)
+            {
+                OnReceiveError(this, CHIP_ERROR_INBOUND_MESSAGE_TOO_BIG, nullptr);
+            }
+        }
+    }
+}
+
+CHIP_ERROR UDPEndPointImplOT::SetMulticastLoopback(IPVersion aIPVersion, bool aLoopback)
+{
+    (void) aIPVersion;
+    (void) aLoopback;
+    // TODO
+    return CHIP_NO_ERROR;
```

should we return ERR_NOT_IMPLEMENTED here and below rather than silent do nothing?

If not, a comment explaining would help. `TODO` does not explain why NO_ERROR was  chosen.

---

Set to ERR_NOT_IMPLEMENTED